lib/matplotlib/axes/_axes.py

```diff
@@ -4559,6 +4559,16 @@ def scatter(self, x, y, s=None, c=None, marker=None, cmap=None, norm=None,
             collection.set_cmap(cmap)
             collection.set_norm(norm)
             collection._scale_norm(norm, vmin, vmax)
+        else:
+            extra_kwargs = {
+                    'cmap': cmap, 'norm': norm, 'vmin': vmin, 'vmax': vmax
+                    }
+            extra_keys = [k for k, v in extra_kwargs.items() if v is not None]
+            if any(extra_keys):
+                keys_str = ", ".join(f"'{k}'" for k in extra_keys)
+                _api.warn_external(
+                        "'c' is empty or not given, "
```

Comment:
Not sure if "empty" is the right word here since `c=[]` is valid.

Reply:
From my understanding `c=[]` is valid if `x` and `y` are also `[]`. But if x and y have values and `c=[]`, `cmap` and the others won't be used.